### DISCUSSION TITLE
Class-based controllers

### DISCUSSION THREAD

**Discussion by gonzacastro** [fastapi/fastapi#6235] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class ItemCBV:
    session: Session = Depends(get_db)
    user_id: UserID = Depends(get_jwt_user)

    @router.post("/item")
    def create_item(self, item: ItemCreate) -> ItemInDB:
        item_orm = ItemORM(name=item.name, owner=self.user_id)
        self.session.add(item_orm)
        self.session.commit()
        return ItemInDB.from_orm(item_orm)

    @router.get("/item/{item_id}")
    def read_item(self, item_id: ItemID) -> ItemInDB:
        item_orm = get_owned_item(self.session, self.user_id, item_id)
        return ItemInDB.from_orm(item_orm)
```

### Description

Hi, I wanted to know if, at the time of defining routers and having all controllers methods scattered all around the file, there's some way to gather up all those methods and wrap them into a class so even if I have common attributes like a Service class or a DAO class (or a session or user_id), inject it as a class attribute like in the example code I provided. I researched about it and found that there's a dependency package called fastapi-utils which satisfies my question but I want to do it without it. Thanks

### Operating System

Windows

### FastAPI Version

0.78.0

### Python Version

3.8.10

**Comment by JarroVGIT** [comment]
Well, the fact that there is a package that extends FastAPI with the exact behaviour that you want, pretty much tells you that it is hard to do it without that package. To my knowledge, the way you are describing it is not how FastAPI works currently. 

If you want to apply dependencies to several routes in a router, you can add the dependency to the router itself rather than all the routes. Does that help your use case? 

(I don't really understand why you would want something like this, to be honest, so I am a little out of my depth here, my apologies.)

**Comment by gonzacastro** [comment]
I put a sample code I found on the internet, but the real use case comes with a router associated to Users for example, there in all the endpoints controllers I would have to inject a UserService class in each one of them to handle businness logic and keep the controller cohesion. What I wanted to avoid is writing over and over again the same service in all the methods.
Now it'd be like this:

```
@router.post("/users")
def create_user(user: UserCreate, service: UserService = Depends()) :
        return service.create_user(user)

@router.get("/users")
def get_all_users(service: UserService = Depends()):
        return service.get_all_users()
```
Like that router I have tons more for each object in the domain of my application, so each one has its own service class and in each endpoint there'd be the same service over and over again.

**Comment by JarroVGIT** [comment]
I see, I don't see any way how to achieve that with pure FastAPI, so I would recommend either copying the relevant code of `fastapi-utils` or just use that package as a whole. I have no experience with it by the way :)

**Comment by gonzacastro** [comment]
Ok then, well thank you very much for taking your time to reply!
